packages/gatsby-theme-docs/gatsby-node.js

```diff
@@ -147,34 +175,64 @@ exports.createPages = async ({ graphql, actions, reporter }) => {
   if (navigationYamlResult.errors) {
     reporter.panicOnBuild('🚨  ERROR: Loading "allNavigationYaml" query');
   }
-  const pages = allMdxPagesResult.data.allMdx.edges;
+  const pages = allMdxPagesResult.data.contents.nodes;
   const navigationPages = navigationYamlResult.data.allNavigationYaml.nodes.reduce(
     (pageLinks, node) => [...pageLinks, ...(node.pages || [])],
     []
   );
-  pages.forEach(({ node }) => {
+  pages.forEach(({ childMdx, name }) => {
+    if (name === 'releases') {
+      reporter.panicOnBuild(
+        '🚨  ERROR: file named "releases" not allowed in src/content directory'
+      );
+
+      return;
+    }
+
     const matchingNavigationPage = navigationPages.find(
       (page) =>
-        trimTrailingSlash(page.path) === trimTrailingSlash(node.fields.slug)
+        trimTrailingSlash(page.path) === trimTrailingSlash(childMdx.fields.slug)
     );
     actions.createPage({
       // This is the slug you created before
       // (or `node.frontmatter.slug`)
-      path: node.fields.slug,
+      path: childMdx.fields.slug,
       // This component will wrap our MDX content
       component: require.resolve('./src/templates/page-content.js'),
       // You can use the values in this context in
       // our page layout component
       context: {
-        ...node.fields,
+        ...childMdx.fields,
         shortTitle: matchingNavigationPage
           ? matchingNavigationPage.title
           : undefined,
       },
     });
   });
+
+  createAllReleaseNotesPages(
+    actions,
+    allMdxPagesResult.data.releaseNotes.nodes
+  );
 };
 
+function createAllReleaseNotesPages(actions, nodes) {
```

Comment:
Sorry but this function is really unnecessary, and so are the arguments passed, since it’s defined as a closure.
Please keep it simple and render the `forEach` only.

Reply:
I got tricked by the diff in the phone sorry, I see now the function is moved outside.
Still, I don't see a reason for extracting this part into a function as we have all the other logic of the `createPages` function in one place.